### PR TITLE
add uuid field to courses

### PR DESCRIPTION
**Pull request by SamJamCul**
## Context

Adding a uuid column to the courses table

## Changes proposed in this pull request

Data migration that adds a uuid column

## Link to Trello card

https://trello.com/c/MFIi8aj1/3628-ensure-courses-we-get-from-ttapi-are-unique

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -576,8 +577,8 @@
   create_table "provider_users_providers", force: :cascade do |t|
     t.bigint "provider_id", null: false
     t.bigint "provider_user_id", null: false
-    t.datetime "created_at", default: -> { "now()" }, null: false
-    t.datetime "updated_at", default: -> { "now()" }, null: false
+    t.datetime "created_at", default: -> { "CURRENT_TIMESTAMP" }, null: false
```
Comment (cwrw) [inline db/schema.rb:580]: Could you remove these changes as they are unrelated?

Comment (SamJamCul) [reply db/schema.rb:580]: 👍

**Review by cwrw — APPROVED** [review]
